src/core/lib/channel/metrics.h

```diff
@@ -97,6 +104,27 @@ class GlobalInstrumentsRegistry {
       absl::string_view unit, absl::Span<const absl::string_view> label_keys,
       absl::Span<const absl::string_view> optional_label_keys,
       bool enable_by_default);
+  static GlobalUInt64GaugeHandle RegisterUInt64Gauge(
+      absl::string_view name, absl::string_view description,
+      absl::string_view unit, absl::Span<const absl::string_view> label_keys,
+      absl::Span<const absl::string_view> optional_label_keys,
+      bool enable_by_default);
+  static GlobalDoubleGaugeHandle RegisterDoubleGauge(
+      absl::string_view name, absl::string_view description,
+      absl::string_view unit, absl::Span<const absl::string_view> label_keys,
+      absl::Span<const absl::string_view> optional_label_keys,
+      bool enable_by_default);
+  static GlobalCallbackUInt64GaugeHandle RegisterCallbackUInt64Gauge(
```

Comment:
Note that it seems like OTel's asynchronous Gauge API only support `int64_t` integral type. CC @yashykt

Reply:
Oh, interesting.  I guess that makes sense, since a gauge might want to be negative, unlike a counter.

Any reason we shouldn't make the same change in this API?

Reply:
Changed the gauge APIs to use `int64_t` instead of `uint64_t`.